File: androidApp/src/main/java/com/sbellanger/weather/android/presentation/WeatherDescriptionView.kt
package com.sbellanger.weather.android.presentation

import androidx.compose.foundation.layout.*
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import java.util.*

@Composable
fun WeatherDescriptionView(
    temperature: String,
    city: String,
    date: String,
    description: String,
    humidity: String,
    pressure: String,
    wind: String
) {
    Column(
        modifier = Modifier
            .fillMaxSize(),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        Text(
            text = temperature,
            modifier = Modifier
                .fillMaxWidth()
                .wrapContentSize(Alignment.Center),
            color = Color.White,
            fontWeight = FontWeight.Bold,
            fontSize = 50.sp
        )
        Text(
            text = city,
            modifier = Modifier
                .fillMaxWidth()
                .wrapContentSize(Alignment.Center)
                .padding(top = 20.dp),
            color = Color.White,
            fontWeight = FontWeight.Bold,
            fontSize = 40.sp
        )
        Text(
            text = date,
            modifier = Modifier
                .fillMaxWidth()
                .wrapContentSize(Alignment.Center)
                .padding(top = 16.dp),
            color = Color.Gray,
            fontSize = 20.sp
        )
        Text(
            text = description,
            modifier = Modifier
                .fillMaxWidth()
                .wrapContentSize(Alignment.Center)
                .padding(top = 16.dp),
            color = Color.White,
            fontWeight = FontWeight.Bold,
            fontSize = 20.sp,
        )
        WeatherStatisticsView(
            humidity = humidity,
            pressure = pressure,
            wind = wind
        )
    }
}

File: iosApp/iosApp/Feature/WeatherDescriptionView.swift
//
//  WeatherDescriptionView.swift
//  iosApp
//
//  Created by Stephen BELLANGER on 09/03/2022.
//  Copyright © 2022 orgName. All rights reserved.
//

import SwiftUI

struct WeatherDescriptionView : View {
    let temperature: String
    let city: String
    let date: String
    let description: String
    let humidity: String
    let pressure: String
    let wind: String
    
    var body: some View {
        VStack {
            if let temperature = temperature{
                Text(temperature)
                    .font(.system(size: 36))
                    .bold()
                    .foregroundColor(.white)
            }
            if let city = city{
                Text(city)
                    .font(.system(size: 26))
                    .bold()
                    .foregroundColor(.white)
                    .padding(.top, 16)
            }
            if let date = date{
                Text(date)
                    .font(.system(size: 16))
                    .bold()
                    .foregroundColor(.gray)
                    .padding(.top, 8)
            }
            if let description = description {
                Text(description)
                    .font(.system(size: 16))
                    .bold()
                    .padding(.top, 12)
                    .foregroundColor(.white)
            }
            HStack(alignment: .center, spacing: 20) {
                if let humidity = humidity {
                    WeatherStatistic(text: humidity, iconRes: "ic_humidity")
                }
                if let pressure = pressure {
                    WeatherStatistic(text: pressure, iconRes: "ic_pressure")
                }
                if let wind = wind {
                    WeatherStatistic(text: wind, iconRes: "ic_wind")
                }
            }
            .frame(maxWidth: .infinity)
            .padding(.top, 20)
        }
        .frame(maxWidth: .infinity)
    }
}
